File: exs/mundo_1/c#/017.cs
/*
Desafio 017

Problema: Faça um programa que leia o comprimento do cateto oposto (co) e 
          do cateto adjacente (ca) de um triângulo retângulo, calcule e mostre 
          o comprimento da hipotenusa

Resolução do problema:
*/

using System;

static class io {
    public static string input(string text) {
        Console.Write(text);
        return Console.ReadLine();
    }
    
    public static void print(object data, string end="\n") {
        string text = data.ToString();

        Console.Write($"{text}{end}");
    }
}

class Program {
    public static void Main() {
        double oposto = double.Parse(
          io.input("Informe o cateto oposto: ")
        );

        double adjacente = double.Parse(
          io.input("Informe o cateto adjacente: ")
        );

        double hipotenusa = Math.Sqrt(
          Math.Pow(oposto, 2) + Math.Pow(adjacente, 2)
        );

        io.print($"Hipotenusa {hipotenusa}");
    }
}

File: exs/mundo_1/rust/017.rs
/*
Desafio 017

Problema: Faça um programa que leia o comprimento do cateto oposto e
          do cateto adjacente de um triângulo retângulo. Calcule e
          mostre o comprimento da hipotenusa.
*/

fn main() {
    let mut co = String::new();
    let mut ca = String::new();

    println!("Informe o comprimento do cateto oposto: ");
    std::io::stdin().read_line(&mut co).unwrap();

    println!("Informe o comprimento do cateto adjacente: ");
    std::io::stdin().read_line(&mut ca).unwrap();

    let co = co
        .trim()
        .parse::<f32>()
        .expect("Cateto oposto não é um número");

    let ca = ca
        .trim()
        .parse::<f32>()
        .expect("Cateto adjacente não é um número");

    let hipotenusa = co.hypot(ca);

    println!("{}", hipotenusa);
}
